provider/slof.py

```diff
@@ -0,0 +1,159 @@
+import re
+import time
+import logging
+
+
+def get_boot_content(vm, start=0, start_str='SLOF',
```

Comment:
I have checked the test case,  the most detailed check points is like this:
```
......
Trying to load:  from: /pci@800000020000000/scsi@4/disk@100000000000000 ...   Successfully loaded
......
Notice:
(1)the number behind "scsi@" (for example "4" above) should match the pci address of the scsi controller that this scsi disk attached to
(2)no any errors from output.
```
Not very clear from my point of view, but I am assuming these functions are trying to get some blocks information from the serial log.

Not sure if you can wait until all the information is logged, and read all the file as a string or list, then you can check:
1. no errors in it 
2. key words like `/pci@800000020000000/scsi@4/disk@100000000000000` in like.
It same easier than read line one by one.

Reply:
As we talked, @hereischen agreed with my opinions.